llvm/lib/Transforms/Vectorize/VPlan.h

```diff
@@ -3884,10 +3887,10 @@ class VPlan {
   /// whether to execute the scalar tail loop or the exit block from the loop
   /// latch.
   const VPBasicBlock *getMiddleBlock() const {
-    return cast<VPBasicBlock>(getVectorLoopRegion()->getSingleSuccessor());
+    return cast<VPBasicBlock>(getScalarPreheader()->getSinglePredecessor());
```

Comment:
This restricts to use of getMiddleBlock() to before bypassing guards are introduced?

Reply:
Works as long as the scalar preheader has a single predecessor, i.e., until runtime guards are introduced as additional predecessors.

Reply:
Yep, this will need some extra work for https://github.com/llvm/llvm-project/pull/114292, which I plan to land after this PR.